wagtail/contrib/settings/registry.py

```diff
@@ -67,12 +67,17 @@ def construct_edit_url(self, instance):
 
 
 class Registry(list):
-    def register(self, model, **kwargs):
+    def __init__(self):
+        self._model_icons = {}
+
+    def register(self, model, icon="cog", **kwargs):
```

Comment:
Not sure if this was intended but I just want to flag that the original `header_icon` uses `cogs` (not `cog`), which are different icons. Although, the `SettingMenuItem` does use the `cog` icon, so there's a bit of inconsistency there.

I don't mind either way, though I think `cog` probably fits better here, and if we eventually add something like a "settings index view", that view can use the `cogs` icon instead.

Reply:
I agree, it makes more sense for the default editing / single item in the menu to be a single cog, the parent menu item for settings (plural) is still the cogs one.

Hence, keeping this change as is.